homeassistant/components/prometheus/__init__.py

```diff
@@ -287,6 +296,23 @@ def _handle_climate(self, state):
             )
             metric.labels(**self._labels(state)).set(current_temp)
 
+        current_action = state.attributes.get(ATTR_HVAC_ACTION)
+        if current_action:
+            values = {
+                CURRENT_HVAC_OFF: 0,
+                CURRENT_HVAC_HEAT: 1,
+                CURRENT_HVAC_COOL: 2,
+                CURRENT_HVAC_DRY: 3,
+                CURRENT_HVAC_IDLE: 4,
+                CURRENT_HVAC_FAN: 5,
+            }
+            metric = self._metric(
+                "action",
+                self.prometheus_cli.Gauge,
```

Comment:
This doesn't make any sense. You can't graph this on a gauge. We shouldn't publish this as a number metric.

Reply:
An alternative would be to export state as a metric label, with metric value being 0 or 1 depending on whether a given state is active. So for a given entity there will always be only one state with the value 1, and all other states with the value of 0. Like this:

```
# HELP current_action Current HVAC action
# TYPE current_action gauge
current_action{domain="climate",entity="climate.bedroom",friendly_name="Bedroom",action="off"} 1.0
current_action{domain="climate",entity="climate.bedroom",friendly_name="Bedroom",action="heating"} 0.0
current_action{domain="climate",entity="climate.bedroom",friendly_name="Bedroom",action="cooling"} 0.0
current_action{domain="climate",entity="climate.bedroom",friendly_name="Bedroom",action="drying"} 0.0
current_action{domain="climate",entity="climate.bedroom",friendly_name="Bedroom",action="idle"} 0.0
current_action{domain="climate",entity="climate.bedroom",friendly_name="Bedroom",action="fan"} 0.0
current_action{domain="climate",entity="climate.bathroom",friendly_name="Bathroom",action="off"} 0.0
current_action{domain="climate",entity="climate.bathroom",friendly_name="Bathroom",action="heating"} 1.0
current_action{domain="climate",entity="climate.bathroom",friendly_name="Bathroom",action="cooling"} 0.0
current_action{domain="climate",entity="climate.bathroom",friendly_name="Bathroom",action="drying"} 0.0
current_action{domain="climate",entity="climate.bathroom",friendly_name="Bathroom",action="idle"} 0.0
current_action{domain="climate",entity="climate.bathroom",friendly_name="Bathroom",action="fan"} 0.0
```

This is a more conventional way to do this in Prometheus, however will involve exporting 6 metrics per entity. This could be a bit confusing, since I suspect most entities do not actually support all 6 possible states (so we will be exporting a metric with a value of 0 that can never have a value of 1 for a given entity).

Would you prefer this over encoding the state in the value? Or is there a better alternative you have in mind?

Reply:
I don't know anything about Promotheus or what best practices would be. Just know that you can't convert an enum into an int 🙂

Reply:
Both approaches – encoding state in the value, or as a separate label – are common in Prometheus. For example, if you look at the `node_exporter` (Prometheus binary for exporting machine-level metrics) you would see the former used for [runit service state](https://github.com/prometheus/node_exporter/blob/2477c5c67dff7e7655a9d466450235e9c9eac193/collector/runit.go#L87) and the latter for [systemd service state](https://github.com/prometheus/node_exporter/blob/6ad94ae4bc5527037ce0e6b8462f58cb3b8ccef3/collector/systemd_linux.go#L254), for example.

Since you seem to be opposed to encoding state as a metric value, I have amended this change to use the label approach as described in my previous comment.

Could you please take another look?

Reply:
Hey @balloob, just wanted to ask if you are happier with the new revision and whether you have any other comments I can address. Thanks!

Reply:
I think that this should be reviewed by someone who understands Prometheus, which is not me.

Reply:
Could you please suggest someone specific?

Reply:
@balloob, ping? Could you please review this or nominate a reviewer?

Reply:
I don't know.